pkgs/development/libraries/science/math/sympow/default.nix

```diff
@@ -0,0 +1,82 @@
+{ stdenv
+, fetchurl
+, fetchpatch
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sympow";
+  version = "1.018.1";
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    # Original website no longer reachable
+    url = "http://mirrors.mit.edu/sage/spkg/upstream/sympow/sympow-${version}.tar.bz2";
+    sha256 = "0hphs7ia1wr5mydf288zvwj4svrymfpadcg3pi6w80km2yg5bm3c";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  configurePhase = ''
+    ./Configure # doesn't take any options
```

Comment:
please add ```runHook preConfigure``` and ```runHook postConfigure```.

Reply:
Isn't that only needed when `preConfigure` or `postConfigure` are actually used? Hooks don't use those right?

Reply:
If someone needs to override something in a package, that can be done easily through the pre/post hooks. If you don't run them when you override a phase, they cannot be used.

Reply:
Alright that makes sense. Is there a reason `configurePhase` etc. are not defined somewhat like this:

```
configurePhase() { 
	runHook preConfigure
	if [[ -n "$configurePhase" ]]; then
		eval "$configurePhase";
	else
		# what the default configure phase does
	fi
	runHook postConfigure
}
```

That would remove this necessary boiler plate and would remove confusion about how to execut phases when using `nix-shell.

Reply:
I'm sure there's a reason - I just don't know what it is! Your suggestion makes perfect sense to me but @vcunat and/or @Ericson2314 might be able to shed some light on the current state of affairs.

Reply:
I'm not sure.  It's probably just because of history; I suspect it comes from beginnings of nixpkgs.  In time it turned out you usually only want to override the inside and not pre/post, and now it's much more difficult to change such basic things.